nemo/collections/nlp/models/language_modeling/megatron_gpt_sft_model.py

```diff
@@ -638,19 +732,36 @@ def cast_for_metric(self, pred, label, metric_name, class_labels=None, labels_ar
         return pred, label
 
     # Override the parent batch reconfiguring logic.
-    def _reconfigure_and_process_inference_batch(self, batch):
-        global_batch_per_gpu = batch['tokens'].size(0)
-        # This should happen only on the last batch of the validation/test dataset with drop_last=False.
-        if global_batch_per_gpu != self.cfg.data.validation_ds.global_batch_size:
-            app_state = AppState()
-            _reconfigure_microbatch_calculator(
-                rank=app_state.global_rank,
-                rampup_batch_size=None,
-                global_batch_size=global_batch_per_gpu * parallel_state.get_data_parallel_world_size(),
-                micro_batch_size=global_batch_per_gpu,
-                data_parallel_size=parallel_state.get_data_parallel_world_size(),
-            )
-        return batch
+    def _reconfigure_and_process_inference_batch(self, batch, data_cfg):
```

Comment:
@stevehuang52 do you remember whether and why this change is necessary?

Reply:
it's from the source PR https://github.com/NVIDIA/NeMo/pull/6915